definitions/sources/declares.js

```diff
@@ -0,0 +1,26 @@
+const staging_tables = ["pages", "requests", "parsed_css"]
+for (const table of staging_tables) {
+    declare({
+        schema: "crawl_staging",
+        name: table,
+    });
+}
+
+const crux_tables = ["country_summary", "device_summary"];
+const past_month = constants.fn_past_month(constants.current_month).substring(0, 7).replace("-", "");
+for (const table of crux_tables) {
+    declare({
+        database: "chrome-ux-report",
+        schema: "materialized",
+        name: table,
+    });
+
+    assert(`${table}_not_empty`).query(ctx => `
+SELECT 
+    'No data for the specified date' AS error_message
+FROM ${ctx.ref("chrome-ux-report", "materialized", table)}
+WHERE yyyymm = ${past_month}
+GROUP BY yyyymm
+HAVING COUNT(1) = 0
+    `);
+}
```

Comment:
Should we add similar asserts for HTTP Archive table?

It _should_ always be done before CrUX but know from bitter experience when the last pipeline broke that that isn't always the case.

Reply:
We could, for `crawl_staging` :)
What would you usually check for?

What was not done before CrUX - pipeline ran before CrUX data became available?
Was the issue with the trigger?

Reply:
This comes back to ,my other question as to when these asserts is run. But I presumed it was used to stop the core_web_vitals/technology query running too early (As that's the only one in this pipeline using CrUX)? It needs before the HTTP Archive data and the CrUX data so shouldn't we assert that both are available before running it?

Reply:
It's running as a dependency of `technologies` table - see [graph visualization](https://console.cloud.google.com/bigquery/dataform/locations/us-central1/repositories/crawl-data/workspaces/dev/actions?authuser=7&project=httparchive).
So if they fail - the rest of the workflow will be cancelled.

My understanding was like this:
1. we trigger this pipeline using a crawl start event (let say August crawl)
2. crawl uses the list of origins from the CrUX for the previous month - `chrome-ux-report.all.202407`
3. that's why we check availability of CrUX dependencies of cwv tech report (there is a couple of hours gap befor all materialized tables become available)
4. we don't need to verify `all.pages` availability because it is supposed to be available since previous "crawl_completed" event somewhere mid July

So I would expect the next run to happen like this:
- next cwv tech report workflow run on ~Sep 13 when we start a crawl
- the CrUX materialized.*_summary tables yyyymmdd=20240801 become available shortly before that
- the `all.pages` date='2024-08-01' data is available since Aug 25

Reply:
Yeah I guess it's much less likely now the BigQuery upload is part of the crawl itself. Previously it was a completely separate job (crawl write HAR files, pipeline read HAR files and inserted data to BigQuery) and so it was possible for the next crawl to start, before the previous upload ran (and it did on occasion!). OK let's leave as is.

Reply:
Good.
If the above explanation is OK - lets run it and watch for failing workflows notifications.